src/util/key_reader.py

```diff
@@ -0,0 +1,99 @@
+from config import LIST_ARDUINO_SERIAL_DEVICES_PATH
+from src.util.logger import get_logger
+import subprocess
+import serial
+import re
+
+logger = get_logger()
+
+class NoReaderFound(Exception):
+    pass
+
+def abstract(fun):
+    def abstract_wrapper(*args, **kwargs):
+        raise NotImplementedError("This is only and abstract function")
+    return abstract_wrapper
+
+class KeyReader(object):
```

Comment:
My thought is that we might have several implementations of key readers (Aptus, HID devices, EM4100 etc.), which is what this interface class should abstract.

Reply:
Sounds good.